test/index.js

```diff
@@ -229,40 +259,51 @@ describe('global-proxy', function() {
     });
     afterEach(function() {
       localSandbox.restore();
+      localSandbox = null;
     });
 
     it('(got proxying set up)', function() {
       assert.isTrue(globalTunnel.isProxying);
     });
 
     describe('with the request library', function() {
-      it('will proxy http requests', function(done) {
+      it('will' + (testParams.proxyEnableFunction === false ? ' not' : '') + ' proxy http requests', function(done) {
```

Comment:
Tests check for the existence of the option, not for its return value depending on the parameters it received